### PR TITLE
 Add a simple progress bar feature 

### PR DESCRIPTION
**Pull request by thantez**
I made a progress bar that can show progress of checked subnets and checked IPs. For this feature, It was necessary to handle 10th argument in `fncCheckSubnet` function and remove logs about failure or timeout of IPs.

### CODE REVIEW

**Comment by thantez** [comment]
Is it ok or I should modify dev branch too?

File: scripts/cfFindIP.sh [hunk 1]
```diff
@@ -50,15 +50,22 @@ configDir="$scriptDir/../config"
 
 configId="NULL"
 configHost="NULL"
+configPort="NULL"
 configPath="NULL"
 configServerName="NULL"
 
+bar_char_done="#"
```
Comment (MortezaBashsiz) [inline scripts/cfFindIP.sh:57]: Variable names should be like
barCharDone
oneTwoThree

File: scripts/cfFindIP.sh [hunk 2]
```diff
@@ -79,18 +86,40 @@ if [ ! -d "$configDir" ]; then
     mkdir -p "$configDir"
 fi
 
+# Progress bar maker function
+# Base on https://www.baeldung.com/linux/command-line-progress-bar
+function show_progress {
```
Comment (MortezaBashsiz) [inline scripts/cfFindIP.sh:91]: Function name and description should be like
fncShowProgress

File: scripts/cfFindIP.sh [hunk 3]
```diff
@@ -79,18 +86,40 @@ if [ ! -d "$configDir" ]; then
...
-	configServerName="$7"
-	osVersion="$8"
+	resultFile="$3"
+	scriptDir="$4"
+	configId="$5"
+	configHost="$6"
+	configPort="$7"
+	configPath="$8"
+	configServerName="$9"
+	osVersion="${10}"
```
Comment (MortezaBashsiz) [inline scripts/cfFindIP.sh:122]: osVersion="$10"
This is not expected to be string with space

File: scripts/cfFindIP.sh [hunk 4]
```diff
@@ -79,18 +86,40 @@ if [ ! -d "$configDir" ]; then
...
 	local ipList scriptDir resultFile timeoutCommand domainFronting
 	ipList="$1"
-	resultFile="$2"
-	scriptDir="$3"
-	configId="$4"
-	configHost="$5"
-	configPath="$6"
-	configServerName="$7"
-	osVersion="$8"
+	resultFile="$3"
```
Comment (MortezaBashsiz) [inline scripts/cfFindIP.sh:115]: Where is $2 ????

File: scripts/cfFindIP.sh [hunk 5]
```diff
@@ -184,16 +207,20 @@ do
 		echo "will use local file"
 		cloudFlareIpList=$(cat "$scriptDir"/cf.local.iplist)
```
Comment (MortezaBashsiz) [inline scripts/cfFindIP.sh:208]: This variable is just list and is not total IP count

**Review by MortezaBashsiz — CHANGES_REQUESTED** [review]
Please review

File: scripts/cfFindIP.sh (hunk 4 above)
Comment (thantez) [reply scripts/cfFindIP.sh:115]: The second argument is progress bar. I used Parallel's built-in progress bar (--bar) that show parameters values as a text in the progress bar. So first argument is IP and second is progress bar text and other arguments are followed.

File: scripts/cfFindIP.sh (hunk 5 above)
Comment (thantez) [reply scripts/cfFindIP.sh:208]: You're right, but handling of shared source of threads is an extra work for this project. I just made a total progress bar that shows progress of checked main IPs and then about subnets I preferred to use Parallel's progress bar, this means we have two progress bars that first one (that's about total IPs not subnets) is handled by script and second is not handled.

File: scripts/cfFindIP.sh (hunk 3 above)
Comment (thantez) [reply scripts/cfFindIP.sh:122]: $10 can't be an argument
see this: https://stackoverflow.com/q/4912733

File: scripts/cfFindIP.sh (hunk 5 above)
Comment (thantez) [reply scripts/cfFindIP.sh:208]: image

Comment (thantez) [reply scripts/cfFindIP.sh:214]: image

**Comment by thantez** [comment]
I improved some tiny things and now this is a screenshot of the result:

image

It can combine with https://github.com/MortezaBashsiz/CFScanner/pull/31#issue-1579499914 to be more pretty.

**Pull request merged**